iphone/cli/commands/_build.js

```diff
@@ -2286,6 +2290,14 @@ iOSBuilder.prototype.checkIfNeedToRecompile = function checkIfNeedToRecompile()
 			return true;
 		}
 
+		// check if the use RunOnMainThread flag has changed
+		if (this.runOnMainThread !== manifest.runOnMainThread) {
+			this.logger.info(__('Forcing rebuild: use RunOnMainThread flag changed since last build'));
+			this.logger.info('  ' + __('Was: %s', manifest.useJSCore));
+			this.logger.info('  ' + __('Now: %s', this.useJSCore));
```

Comment:
Should be `manifest.runOnMainThread` and `this.runOnMainThread`, not `manifest.useJSCore` and `this.useJSCore`.